## Regresión lineal en python con scikit-learning

In [3]:
from sklearn.feature_selection import RFE
from sklearn.svm import SVR
import pandas as pd
import numpy as np

In [5]:
data = pd.read_csv('datasets/ads/Advertising.csv')

In [7]:
feature_cols =  ['TV', 'Radio','Newspaper']

In [11]:
x = data[feature_cols]
y = data['Sales']

In [12]:
estimator = SVR(kernel='linear')
selector = RFE(estimator, 2, step=1)
selector = selector.fit(x, y)

In [13]:
selector.support_

array([ True,  True, False])

In [14]:
selector.ranking_

array([1, 1, 2])

In [15]:
from sklearn.linear_model import LinearRegression

In [16]:
x_pred = x[['TV','Radio']]

In [17]:
lm = LinearRegression()
lm.fit(x_pred,y)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [19]:
lm.intercept_, lm.coef_

(2.921099912405138, array([0.04575482, 0.18799423]))

* el modelo generado por scikit learning
* sales = 2.921099912405138 +  0.04575482 *TV + 0.18799423*Radio